SampleID, TargetID, Cq

ewallace opened the pull request:
Pull request fixes #39, #40, #42, giving fundamental variables better names:

* SampleID (replaces Sample)
* TargetID (replaces Probe)
* Cq (replaces Ct); MIQE guidelines specify Cq for quantification cycle.

This is a breaking change as previous code will not be compatible.

Request review from @DimmestP before we merge.

DimmestP commented:
I have re-ran Jamie's RPS3 analysis but with this version and it appears to run perfectly fine. I do have a few points in terms of solidifying the importance of TargetID and SampleID in the code.

Should the create_rowkey_4in16() and create_rowkey_8in16_plain() functions demand a TargetID argument? Otherwise you don't realise you've made a mistake until you run plateNorm().

Likewise with the create_colkey_6in24() and SampleID. 

Perhaps you could check both at the same time with the label_plate_rowcol() function?

ewallace commented:
Great points! I added warnings to `label_plate_rowcol()` if any of `SampleID`, `TargetID`, `Type`, are missing.

Also added the variable `coercefactors` to make factor-coercion of `WellR` and `WellC` a default option rather than an inescapable requirement.

DimmestP commented:
Great, Ive checked the new warning and they work!
